packages/x-date-pickers/src/PickersDay/PickersDay.tsx

```diff
@@ -19,11 +19,28 @@ import {
   pickersDayClasses,
 } from './pickersDayClasses';
 
+export interface ExportedPickersDayProps {
+  /**
+   * If `true`, today's date is rendering without highlighting with circle.
+   * @default false
+   */
+  disableHighlightToday?: boolean;
+  /**
+   * If `true`, days outside the current month are rendered:
+   * - if `fixedWeekNumber` is defined, renders days to have the weeks requested.
```

Comment:
Could we look into formatting it so that the new lines are also rendered in the api doc? 🤔 
Also, how does it look like in IDE?

Reply:
![image](https://user-images.githubusercontent.com/3309670/212665178-12041c3e-827c-4c46-93e2-3ae0af32b788.png)

In Webstorm it looks fine

In the API page, it's super ugly like all our props with long descriptions

![image](https://user-images.githubusercontent.com/3309670/212665423-bd36cd4d-aff8-45f8-8aa1-fbc85f3915a9.png)

We should really priority improving this table... But not sure we can do much on this PR.

What do you have in mind ?

Reply:
> What do you have in mind ?

I propose the following formatting:
```js
/**
 * If `true`, days outside the current month are rendered:
 *
 * - if `fixedWeekNumber` is defined, renders days to have the weeks requested.
 *
 * - if `fixedWeekNumber` is not defined, renders day to fill the first and last week of the current month.
 *
 * - if `calendars` is set to `1` on range pickers.
 * @default false
 */
showDaysOutsideCurrentMonth?: boolean;
```

This would produce the following result. 
<img width="571" alt="Screenshot 2023-01-16 at 12 33 10" src="https://user-images.githubusercontent.com/4941090/212679390-352a2bc2-423c-4d10-914b-cf921b1b9c33.png">
<img width="192" alt="Screenshot 2023-01-16 at 12 33 21" src="https://user-images.githubusercontent.com/4941090/212679397-64062b60-6b00-49c9-abad-9a43bbebfb11.png">

Reply:
I don't understand the `if `calendars` is set to `1` on range pickers.` :angel: 
I applied the rest

Reply:
I was trying to unify the text formatting, because that last sentence about range pickers with different formatting seemed off. 🤔 
Adding it as another list item IMHO improves the coherency of the description.

Reply:
Hiow about `- ignored if `calendars equals more than `1` on range pickers.` ?

With your proposal, the two first dots explains have the `if A, then B` but the 3rd is just `if A` so I'm waiting for a 2nd part.
Note that the first two dots also applies to the range pickers.

If it was a doc page it would probably be something like:

```
If `true`, days outside the current month are rendered:

- if `fixedWeekNumber` is defined, renders days to have the weeks requested.

- if `fixedWeekNumber` is not defined, renders day to fill the first and last week of the current month.

:::info
On range pickers, it will be ignored if more than one `calendars` is rendered.
:::
```

Reply:
Fair point. Your suggestion sound more correct. 👌 👍

Reply:
Done :ok_hand: